File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/{{name.snakeCase()}}/Sources/{{name.snakeCase()}}/Domain/UseCase/SyncDataUseCase.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

import Foundation

/// Use case to trigger data synchronization via the repository.
public struct SyncDataUseCase: Sendable {
    private let repository: {{name.pascalCase()}}Repository

    public init(repository: {{name.pascalCase()}}Repository) {
        self.repository = repository
    }

    public func execute() async throws {
        try await repository.syncData()
    }
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/domain/usecase/SyncDataUseCase.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

package com.danhdue.{{name.snakeCase()}}.domain.usecase

import com.danhdue.{{name.snakeCase()}}.domain.repository.{{name.pascalCase()}}Repository
import javax.inject.Inject

class SyncDataUseCase @Inject constructor(
    private val repository: {{name.pascalCase()}}Repository
) {
    suspend fun execute(): Result<Boolean> = repository.syncData()
}
